apps/docs/pages/guides/functions/debugging.mdx

```diff
@@ -11,82 +11,135 @@ export const meta = {
+- **Event Loop Completion**: If you observe "Event loop completed" in your error logs, it's likely your function is not implemented correctly. You should check your function code for any syntax errors, infinite loops, or unresolved promises that might cause this error. Or you can try running the function locally (using Supabase CLI **`functions serve`**) to see if you can debug the error. The local console should give a full stack trace on the error with line numbers of the source code.
 
 ### Issues serving Edge Functions locally with the Supabase CLI
 
-- Make sure you're on the latest version of the [Supabase CLI](/docs/guides/cli#updates).
-- Run the serve command with the `--debug` flag.
+- Make sure you're on the latest version of the [Supabase CLI](/docs/guides/cli#updates).
+- Run the serve command with the `-debug` flag.
 - Support engineers can then try to run the provided sample code locally and see if they can reproduce the issue.
-- Search the [Edge Runtime](https://github.com/supabase/edge-runtime) and [CLI](https://github.com/supabase/cli) repos for the error message, to see if it has been reported before.
+- Search the [Edge Runtime](https://github.com/supabase/edge-runtime) and [CLI](https://github.com/supabase/cli) repos for the error message, to see if it has been reported before.
 - If the output from the commands above does not help you to resolve the issue, please open a support ticket via the Supabase Dashboard (by clicking the "Help" button at the top right) and include all output and details about your commands.
 
-## Advanced Techniques
+## Advanced techniques
 
-### Checking Function Boot Time
+### Monitoring Edge Function Resource Usage
 
-Check the logs for the function. In the logs, look for a "Booted" event and note the reported boot time. If available, click on the event to access more details, including the regions from where the function was served. Investigate if the boot time is excessively high (longer than 1 second) and note any patterns or regions where it occurs.
+To determine how much memory and CPU your edge function consumes, follow these steps:
 
-### Finding Bundle Size
+- Navigate to the Supabase Dashboard.
+- Go to **Edge Functions**.
+- Select the specific function by clicking on its name.
+- View the resource usage **Metrics** on the charts provided.
 
-To find the bundle size of a function, run the following command locally:
+### Resource Allocation
 
-```bash
-deno info /path/to/function/index.ts
-```
+Ensure your project has sufficient resources to handle the traffic. Under-provisioned resources can lead to HTTP timeouts and 5xx response codes. Monitor resource utilization through the Dashboard metrics and consider upgrading if necessary.
+
+### Query and Index Optimization
+
+Fine-tune your database queries and ensure proper indexing. Use `pg_stat_statements` to identify slow queries and optimize them to reduce load on the system.
+
+### Configuring Edge Function Timeouts
```

Comment:
This is webhook timeout, right? We should call it out explicitly otherwise can be confusing as it's the Edge Function duration.